elements/pl-integer-input/pl-integer-input.py

```diff
@@ -37,9 +37,22 @@ def prepare(element_html, data):
         if name in data['correct_answers']:
             raise Exception('duplicate correct_answers variable name: %s' % name)
         # Test conversion, but leave as string so proper value is shown on answer panel
-        if pl.string_to_integer(correct_answer, base) is None:
+        a_tru_parsed = pl.string_to_integer(correct_answer, base)
+        if a_tru_parsed is None:
             raise Exception('correct answer is not a valid input: %s' % name)
+        if a_tru_parsed > 2 ** 53 - 1 or a_tru_parsed < -((2 ** 53) - 1):
+            raise Exception('answer must be between (2**53)-1 and -((2**53)-1)')
         data['correct_answers'][name] = correct_answer
+        return
+
+    a_tru = pl.from_json(data['correct_answers'].get(name, None))
+    if a_tru is not None:
+        if isinstance(a_tru, str):
+            a_tru = pl.string_to_integer(a_tru, base)
+        if a_tru is None:
+            raise Exception('correct answer is not a valid input: %s' % name)
```

Comment:
Should we do `a_tru = int(a_tru)` here to force a conversion? Maybe inside a `try` block? Later we will do this: https://github.com/PrairieLearn/PrairieLearn/blob/f55ea11694ce820c2290be2d6c6a80611c2a0468/elements/pl-integer-input/pl-integer-input.py#L255

Reply:
If i add a try block do I add the except block just to throw another exception but with a more specific error message?

Reply:
That would be great.

Reply:
https://github.com/nitahieb/PrairieLearn/blob/2aa9eaea9927ea2afbeda189ea70c4a9aedb64a9/elements/pl-integer-input/pl-integer-input.py#L48-L56

Reply:
The current (new) version only converts to int if the value was originally a string. I suggest removing the `if instanceof` line and convert to int without checking the original type. The exception would also catch a type that cannot be converted to an int (e.g., a list).